nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/syslog/SyslogReader.java

```diff
@@ -0,0 +1,148 @@
+    public static final String GENERIC_SYSLOG_SCHEMA_NAME = "default-syslog-schema";
+    static final AllowableValue GENERIC_SYSLOG_SCHEMA = new AllowableValue(GENERIC_SYSLOG_SCHEMA_NAME, "Use Generic Syslog Schema",
+            "The schema will be the default Syslog schema.");
+    public static final PropertyDescriptor CHARSET = new PropertyDescriptor.Builder()
+            .name("Character Set")
+            .description("Specifies which character set of the Syslog messages")
+            .required(true)
+            .defaultValue("UTF-8")
+            .addValidator(StandardValidators.CHARACTER_SET_VALIDATOR)
+            .build();
+
+    private volatile SyslogParser parser;
+    private volatile RecordSchema recordSchema;
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        final List<PropertyDescriptor> properties = new ArrayList<>(1);
+        properties.add(CHARSET);
+        return properties;
+    }
+
+
+    @OnEnabled
+    public void onEnabled(final ConfigurationContext context) {
+        final String charsetName = context.getProperty(CHARSET).getValue();
+        parser = new SyslogParser(Charset.forName(charsetName));
+        recordSchema = createRecordSchema();
+    }
+
+    @Override
+    protected List<AllowableValue> getSchemaAccessStrategyValues() {
+        final List<AllowableValue> allowableValues = new ArrayList<>();
+        allowableValues.add(GENERIC_SYSLOG_SCHEMA);
+        return allowableValues;
+    }
+
+    @Override
+    protected AllowableValue getDefaultSchemaAccessStrategy() {
+        return GENERIC_SYSLOG_SCHEMA;
+    }
+
+    @Override
+    protected SchemaAccessStrategy getSchemaAccessStrategy(final String strategy, final SchemaRegistry schemaRegistry, final ConfigurationContext context) {
+        return createAccessStrategy();
+    }
+
+    @Override
+    protected SchemaAccessStrategy getSchemaAccessStrategy(final String strategy, final SchemaRegistry schemaRegistry, final ValidationContext context) {
+        return createAccessStrategy();
+    }
+
+    static RecordSchema createRecordSchema() {
+        final List<RecordField> fields = new ArrayList<>();
+        fields.add(new RecordField(SyslogAttributes.PRIORITY.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.SEVERITY.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.FACILITY.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.VERSION.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.TIMESTAMP.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.HOSTNAME.key(), RecordFieldType.STRING.getDataType(), true));
+        fields.add(new RecordField(SyslogAttributes.BODY.key(), RecordFieldType.STRING.getDataType(), true));
```

Comment:
Can we add the additionalDetails page with the string representation of this schema like we did for the other one? I know this one is simpler, but its nice to have for reference.

Reply:
Sorry, of course, that was an oversight on my part

Reply:
btw, it would be great if we generated the schema documentation the way we do the annotated documentation etc... I wonder how we could do that?

Reply:
We could have an annotation like @ProducesSchema with an array of @Field annotations and then use that in the docs, kind of like the @WritesAttributes... but it would only be useful for the cases where there is a pre-defined schema and we have very few of those right now

Reply:
doc is in